kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -405,44 +403,53 @@ ColumnVector visitAlwaysFalse(AlwaysFalse alwaysFalse) {
         @Override
         ColumnVector visitComparator(Predicate predicate) {
             PredicateChildrenEvalResult argResults = evalBinaryExpressionChildren(predicate);
-
-            int numRows = argResults.rowCount;
-            boolean[] result = new boolean[numRows];
-            boolean[] nullability = evalNullability(argResults.leftResult, argResults.rightResult);
-            int[] compareResult = compare(argResults.leftResult, argResults.rightResult);
+            VectorComparator v;
             switch (predicate.getName()) {
                 case "=":
-                    for (int rowId = 0; rowId < numRows; rowId++) {
-                        result[rowId] = compareResult[rowId] == 0;
-                    }
-                    break;
+                    v = new VectorComparator() {
+                        @Override
+                        public boolean compare(int compareResult) {
+                            return compareResult == 0;
+                        }
+                    };
```

Comment:
can this be defined  as a lambda (compareResult) -> (compareResult == 0)?

```
comparatorVector(leftResult, rightResult,  (compareResult) -> (compareResult == 0))
```

Reply:
good call, fixed.